File: backend/internal/domain/entities/quiz_attempt.go
package entities

import (
	"time"

	"github.com/google/uuid"
)

type QuizAttemptAnswer struct {
	QuestionID string `json:"question_id"`
	AnswerID   string `json:"answer_id"`
}

type QuizAttempt struct {
	ID          uuid.UUID
	QuizID      uuid.UUID
	StudentID   uuid.UUID
	OrgID       uuid.UUID
	Score       int
	MaxScore    int
	Answers     []QuizAttemptAnswer
	SubmittedAt time.Time
	CreatedAt   time.Time
	UpdatedAt   time.Time
}

File: frontend/src/types/quiz_attempt.ts
export interface QuizAttemptResultDTO {
  id: string;
  quiz_id: string;
  score: number;
  max_score: number;
  percentage: number;
  submitted_at: string;
}

export interface QuizAnswerInput {
  question_id: string;
  answer_id: string;
}
